pennylane/drawer/mpldrawer.py

```diff
@@ -842,3 +848,161 @@ def measure(self, layer, wires, box_options=None, lines_options=None):
+                        ),
+                        plt.Line2D(
+                            (start_x, layer - self._cond_shift),
+                            (next_wire - self._cond_shift,) * 2,
+                            **options,
+                        ),
+                        plt.Line2D(
+                            (start_x, layer - self._cond_shift),
+                            (next_wire + self._cond_shift,) * 2,
+                            **options,
+                        ),
+                    )
+                )
+        elif wires_target[-1] < wires_ctrl[0]:
+            lines.extend(
+                (
+                    # draw from bottom-most measurement to double-elbow
+                    plt.Line2D(
+                        (start_x, layer + self._cond_shift),
+                        (wires_ctrl[-1] + self._cond_shift,) * 2,
+                        **options,
+                    ),
+                    plt.Line2D(
+                        (start_x, layer - self._cond_shift),
+                        (wires_ctrl[-1] - self._cond_shift,) * 2,
+                        **options,
+                    ),
+                    # draw vertical lines that reach the target operation
+                    plt.Line2D(
+                        (layer + self._cond_shift,) * 2,
+                        (wires_ctrl[-1] + self._cond_shift, wires_target[-1]),
+                        **options,
+                    ),
+                    plt.Line2D(
+                        (layer - self._cond_shift,) * 2,
+                        (wires_ctrl[0] - self._cond_shift, wires_target[-1]),
+                        **options,
+                    ),
+                )
+            )
+            for wire_idx, ctrl_wire in enumerate(wires_ctrl[:-1]):
+                # draw _  for every wire but the first one
+                #      ‾|
+                lines.extend(
+                    (
+                        plt.Line2D(
+                            (layer - self._cond_shift,) * 2,
+                            (
+                                ctrl_wire + self._cond_shift,
+                                wires_ctrl[wire_idx + 1] - self._cond_shift,
+                            ),
+                            **options,
+                        ),
+                        plt.Line2D(
+                            (start_x, layer - self._cond_shift),
+                            (ctrl_wire - self._cond_shift,) * 2,
+                            **options,
+                        ),
+                        plt.Line2D(
+                            (start_x, layer - self._cond_shift),
+                            (ctrl_wire + self._cond_shift,) * 2,
+                            **options,
+                        ),
+                    )
+                )
+        else:
+            raise ValueError(
+                "Cannot draw interspersed mid-circuit measurements and conditional operations"
```

Comment:
Could we rephrase so that the "success condition" is clear? e.g.., `mid-circuit measurements must happen on consecutive wires and must be above or below all other wires.

Reply:
done!